pyccel/ast/core.py

```diff
@@ -2673,6 +2673,18 @@ def is_ndarray(self):
             return False
         return isinstance(self.dtype, (NativeInteger, NativeBool,
                           NativeReal, NativeComplex))
+    def rename(self, newname):
+        return Variable(self._dtype,newname, rank=self._rank,
+                        allocatable=self._allocatable,
+                        is_stack_array=self._is_stack_array,
+                        is_pointer=self._is_pointer,
+                        is_target=self._is_target,
+                        is_polymorphic=self._is_polymorphic,
+                        is_optional=self._is_optional,
+                        shape=self._shape, cls_base=self._cls_base,
+                        order=self.order, precision=self.precision,
+                        is_argument=self._is_argument,
+                        is_kwonly=self._is_kwonly)
```

Comment:
Why is this function necessary? The `clone` function does the same thing.
If it is necessary then the function needs a new name. We are not renaming a variable but creating another Variable with the same properties (cloning the Variable). It is important that this distinction is clear as renaming something would not invalidate other existing references to the object, but cloning will

Reply:
this function is necessary for fortran printer. as you mentioned `rename` function creates another variable with the same properties in order to do that with the function `clone` the variable properties needs to be copied before passing them to the function. as for the name of the function it should be changed since cloning is other than renaming. I will remove it until it being used in the fortran printer.

Reply:
> the variable properties needs to be copied before passing them

Why do you say that? The clone method should copy all properties unless you provide a different value in a kwarg

Reply:
yes you are right that where i was mistaking, clone function does the same thing